### PR TITLE
fallback to og image if no images

### PR DESCRIPTION
**Pull request by JosiasAurel**
addresses #380

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -95,9 +95,11 @@ app.event("member_joined_channel", execute(joined));
 
 app.event("user_change", execute(userChanged));
 
-app.message(subtype("file_share"), execute(create));
+app.event("message", execute(create));
 
-app.message(noFileCheck, execute(noFile));
+// app.message(subtype("file_share"), execute(create));
```
Comment (grymmy) [inline src/app.js:100]: please add commentary on why these things are disabled - or delete them entirely

File: src/lib/updates.js [hunk 2]
```diff
@@ -31,10 +31,32 @@ export const createUpdate = async (files = [], channel, ts, user, text) => {
...
+      attachments.push(imageUri);
+      break;
+    }
+  }
+
   if ((attachments.length + videos.length) === 0) {
     await Promise.all([
       react("remove", channel, ts, "beachball"),
       react("add", channel, ts, "x"),
+      app.client.chat.delete({
```
Comment (grymmy) [inline src/lib/updates.js:54]: add comment explaining what this does

File: src/lib/updates.js [hunk 3]
```diff
@@ -31,10 +31,32 @@ export const createUpdate = async (files = [], channel, ts, user, text) => {
...
+        token: process.env.SLACK_USER_TOKEN,
+        channel,
+        ts
+      }),
+      postEphemeral(channel, t("messages.delete", { text }), user)
```
Comment (grymmy) [inline src/lib/updates.js:59]: add comment explaining

File: src/lib/utils.js [hunk 4]
```diff
@@ -63,3 +64,47 @@ export const getUrlFromString = (str) => {
   if (url.startsWith("<")) url = url.substring(1, url.length - 1);
   return url;
 };
+
+// returns the urls that are in the text
+export function getUrls(text) {
+  const matcher = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()'@:%_\+.~#?!&//=]*)/gi;
```
Comment (grymmy) [inline src/lib/utils.js:70]: describe at high level what this regex does, if it conforms to a standard, does it support https vs http, etc

File: src/events/reactionAdded.js [hunk 5]
```diff
@@ -48,11 +48,15 @@ export default async ({ event }) => {
         user
       );
     } else if (message) {
+
+      let update;
+
       if (!message.files || message.files.length == 0) {
```
Comment (grymmy) [inline src/events/reactionAdded.js:54]: This is just a general rule of mine - when I see two lines in a conditional that nearly match such as what is contained here, I usually extract the difference into a variable and then condense the code to one call.  For instance, you could refactor this to:

```
let hasNoFiles = !message.files || message.files.length == 0
let files = hasNoFiles ? [] : message.files
let update = await createUpdate(files, channel, ts, user, message.text)
```

I would argue ^ is more readable & concise, also doesn't require the predefined-and-then-mutated variable 'update'

**Review by grymmy — CHANGES_REQUESTED** [review]

File: src/lib/updates.js [hunk 6]
```diff
@@ -31,10 +31,34 @@ export const createUpdate = async (files = [], channel, ts, user, text) => {
     }),
   ])
 
+  // if there are no attachments, attempt to get from the first link having an og image 
+  if (!text) return;
+  const urls = getUrls(text);
+  if (urls) {
+    for (const url of urls) {
```
Comment (grymmy) [inline src/lib/updates.js:38]: nit - could this be parallelized, or is it truly order dependent?

**Review by grymmy — APPROVED** [review]

**Comment by MatthewStanciu** [comment]
Woah cool!!! Thanks for doing this!

**Pull request merged**